src/NoobFlexibleStaking.sol

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "forge-std/Test.sol";
+import "forge-std/console.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/Pausable.sol";
+
+contract NoobFlexibleStaking is Ownable, ReentrancyGuard, Pausable {
+    using SafeERC20 for IERC20;
+
+    /// <=============== Events ===============>
+    event Staked(address indexed user, uint256 amount);
+    event Claimed(address indexed user, uint256 rewards);
+    event Unstaked(address indexed user, uint256 amount);
+    event AprUpdated(uint256 newApr);
+
+    /// <=============== STATE VARIABLES ===============>
+    IERC20 public noobToken;
+
+    // Struct to hold user's stake details
+    struct Stake {
+        uint256 amount; // Total staked amount
+        uint256 lastClaimedAt; // Last time rewards were claimed or updated
+        uint256 rewards; // Accumulated rewards
+        uint256 lastApr; // Last APR used for reward calculation
+    }
+
+    // Mapping to keep track of user stakes
+    mapping(address => Stake) public userStakes;
```

Comment:
We might need to support multiple stakes per user

Reply:
why?

Reply:
@Ziden I initially thought that when a user stakes for a second time, the staking amount should be topped up, and previous rewards could be added to the staking amount as well. Please correct me if I'm wrong.
Alternatively, I can modify the logic so that each user will have multiple stakes.

Reply:
Hey JC ! If the topup works without duplicating the stake which was my initial fear then it's fine.

I was chatting with Corby about this yesterday i believe he wrote some tests to cover those scenarios.